extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSources.java

```diff
@@ -149,6 +152,40 @@ public AgroalDataSource doCreateDataSource(String dataSourceName) {
                     "Unable to load the datasource driver " + resolvedDriverClass + " for datasource " + dataSourceName, e);
         }
 
+        if (dataSourceJdbcBuildTimeConfig.tracing) {
+            boolean tracingEnabled = dataSourceJdbcRuntimeConfig.tracing.enabled.orElse(dataSourceJdbcBuildTimeConfig.tracing);
+
+            if (tracingEnabled) {
+                StringBuilder tracingURL = new StringBuilder(
+                        dataSourceJdbcRuntimeConfig.url.get().replace("jdbc:", "jdbc:tracing:"));
+
+                if (dataSourceJdbcRuntimeConfig.tracing.traceWithActiveSpanOnly) {
+                    if (!tracingURL.toString().contains("?")) {
+                        tracingURL.append("?");
+                    }
+
+                    tracingURL.append("traceWithActiveSpanOnly=true");
+                }
+
+                if (dataSourceJdbcRuntimeConfig.tracing.ignoreForTracing.isPresent()) {
+                    if (!tracingURL.toString().contains("?")) {
+                        tracingURL.append("?");
+                    }
+
+                    Arrays.stream(dataSourceJdbcRuntimeConfig.tracing.ignoreForTracing.get().split(";"))
+                            .filter(query -> !query.isEmpty())
+                            .forEach(query -> tracingURL.append("ignoreForTracing=")
+                                    .append(query.replaceAll("\"", "\\\""))
+                                    .append(";"));
+                }
+
+                dataSourceJdbcRuntimeConfig.url = Optional.of(tracingURL.toString());
```

Comment:
Sorry I missed it but please don't write to the config object directly, create a local variable for the URL.